Tests fail on following:

```
Test project /home/imincik/Projects/nix/nixpkgs/code/source/build/tiledb/test
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
    Start 1: tiledb_unit

1: Test command: /home/imincik/Projects/nix/nixpkgs/code/source/build/tiledb/test/tiledb_unit "--durations=yes"
1: Working Directory: /home/imincik/Projects/nix/nixpkgs/code/source/test
1: Test timeout computed to be: 10000000
test 2
    Start 2: test_ci_asserts

2: Test command: /home/imincik/Projects/nix/nixpkgs/code/source/build/tiledb/test/ci/test_assert
2: Working Directory: /home/imincik/Projects/nix/nixpkgs/code/source/build/tiledb/test/ci
2: Test timeout computed to be: 10000000
1: Errors occurred during startup!
1:   Found an unclosed tag while registering test case 'Dimension: Test
1:   map_to_uint64, float32' at /home/imincik/Projects/nix/nixpkgs/code/source/
1:   test/src/unit-dimension.cc:274
1/2 Test #1: tiledb_unit ......................***Failed    0.01 sec
Errors occurred during startup!
  Found an unclosed tag while registering test case 'Dimension: Test
  map_to_uint64, float32' at /home/imincik/Projects/nix/nixpkgs/code/source/
  test/src/unit-dimension.cc:274

2: Assert did not exit!
2: Randomness seeded to: 1241610924
2: retval is 0 (0x0) from /home/imincik/Projects/nix/nixpkgs/code/source/build/tiledb/test/ci/try_assert
2: ===============================================================================
2: All tests passed (1 assertion in 1 test case)
2:
2/2 Test #2: test_ci_asserts ..................   Passed    0.01 sec

50% tests passed, 1 tests failed out of 2

Total Test time (real) =   0.01 sec

The following tests FAILED:
	  1 - tiledb_unit (Failed)
Errors while running CTest
make[7]: *** [test/CMakeFiles/check.dir/build.make:70: test/CMakeFiles/check] Error 8
make[7]: Leaving directory '/home/imincik/Projects/nix/nixpkgs/code/source/build/tiledb'
make[6]: *** [CMakeFiles/Makefile2:13681: test/CMakeFiles/check.dir/all] Error 2
make[6]: Leaving directory '/home/imincik/Projects/nix/nixpkgs/code/source/build/tiledb'
make[5]: *** [CMakeFiles/Makefile2:13688: test/CMakeFiles/check.dir/rule] Error 2
make[5]: Leaving directory '/home/imincik/Projects/nix/nixpkgs/code/source/build/tiledb'
make[4]: *** [Makefile:4261: check] Error 2
make[4]: Leaving directory '/home/imincik/Projects/nix/nixpkgs/code/source/build/tiledb'

make[3]: *** [CMakeFiles/check.dir/build.make:70: CMakeFiles/check] Error 2
make[3]: Leaving directory '/home/imincik/Projects/nix/nixpkgs/code/source/build'
make[2]: *** [CMakeFiles/Makefile2:220: CMakeFiles/check.dir/all] Error 2
make[2]: Leaving directory '/home/imincik/Projects/nix/nixpkgs/code/source/build'
make[1]: *** [CMakeFiles/Makefile2:227: CMakeFiles/check.dir/rule] Error 2
make[1]: Leaving directory '/home/imincik/Projects/nix/nixpkgs/code/source/build'
make: *** [Makefile:189: check] Error 2
```

--- a/pkgs/development/libraries/tiledb/FindMagic_EP.cmake.patch
+++ b/pkgs/development/libraries/tiledb/FindMagic_EP.cmake.patch
@@ -1,0 +1,14 @@
+diff --git a/FindMagic_EP.cmake b/FindMagic_EP.cmake
+--- a/cmake/Modules/FindMagic_EP.cmake
++++ b/cmake/Modules/FindMagic_EP.cmake
+@@ -126,9 +126,7 @@ if(NOT TILEDB_LIBMAGIC_EP_BUILT)
+     # that was modified by tiledb to also build with cmake for nix
+     ExternalProject_Add(ep_magic
+       PREFIX "externals"
+-      GIT_REPOSITORY "https://github.com/TileDB-Inc/file-windows.git"
+-      GIT_TAG "5.38.2.tiledb"
+-      GIT_SUBMODULES_RECURSE TRUE
++      DOWNLOAD_COMMAND true
+       UPDATE_COMMAND ""
+       CMAKE_ARGS
+         -DCMAKE_INSTALL_PREFIX=${TILEDB_EP_INSTALL_PREFIX}

--- a/pkgs/development/libraries/tiledb/default.nix
+++ b/pkgs/development/libraries/tiledb/default.nix
@@ -1,6 +1,7 @@
 { lib
 , stdenv
 , fetchFromGitHub
+
 , cmake
 , zlib
 , lz4
@@ -12,7 +13,7 @@
 , boost
 , libpqxx
 , clang-tools
-, catch2
+, catch2_3
 , python3
 , gtest
 , doxygen
@@ -20,24 +21,54 @@
 , useAVX2 ? stdenv.hostPlatform.avx2Support
 }:
 
+let
+  # pre-fetch ExternalProject from cmake/Modules/FindMagic_EP.cmake
+  ep-file-windows = fetchFromGitHub {
+    owner = "TileDB-Inc";
+    repo   = "file-windows";
+    rev    = "5.38.2.tiledb";
+    hash = "sha256-TFn30VCuWZr252VN1T5NNCZe2VEN3xQSomS7XxxKGF8=";
+    fetchSubmodules = true;
+  };
+
+in
 stdenv.mkDerivation rec {
   pname = "tiledb";
-  version = "2.8.3";
+  version = "2.18.0";
 
   src = fetchFromGitHub {
     owner = "TileDB-Inc";
     repo = "TileDB";
     rev = version;
-    hash = "sha256-HKMVwrPnk9/mukH3mJ2LEAvA9LBF4PcgBZjbbLhO9qU=";
+    hash = "sha256-y5/kXODzkNKmMn8P+WOOWGy9Htd0nhXbUxmfw/SqSYY=";
   };
+
+  patches = [
+    ./FindMagic_EP.cmake.patch
+  ];
+
+  postPatch = ''
+    # copy pre-fetched external project to directory where it is expected to be
+    mkdir -p build/externals/src
+    cp -a ${ep-file-windows} build/externals/src/ep_magic
+    chmod -R u+w build/externals/src/ep_magic
+
+    # add openssl on path
+    sed -i '49i list(APPEND OPENSSL_PATHS "${openssl.dev}" "${openssl.out}")' \
+      cmake/Modules/FindOpenSSL_EP.cmake
+  '';
 
   # (bundled) blosc headers have a warning on some archs that it will be using
   # unaccelerated routines.
   cmakeFlags = [
-    "-DTILEDB_WERROR=0"
+    "-DTILEDB_VCPKG=OFF"
+    "-DTILEDB_WEBP=OFF"
+    "-DTILEDB_WERROR=OFF"
   ] ++ lib.optional (!useAVX2) "-DCOMPILER_SUPPORTS_AVX2=FALSE";
 
   nativeBuildInputs = [
+    ep-file-windows
+    catch2_3
     clang-tools
     cmake
     python3
@@ -49,7 +80,6 @@ stdenv.mkDerivation rec {
   ];
 
   buildInputs = [
-    catch2
     zlib
     lz4
     bzip2
@@ -61,13 +91,14 @@ stdenv.mkDerivation rec {
     libpqxx
   ];
 
-  # emulate the process of pulling catch down
-  postPatch = ''
-    mkdir -p build/externals/src/ep_catch
-    ln -sf ${catch2}/include/catch2 build/externals/src/ep_catch/single_include
-
-    sed -i '38i list(APPEND OPENSSL_PATHS "${openssl.dev}" "${openssl.out}")' \
-      cmake/Modules/FindOpenSSL_EP.cmake
+  # test commands taken from
+  # https://github.com/TileDB-Inc/TileDB/blob/dev/.github/workflows/unit-test-runs.yml
+  checkPhase = ''
+    runHook preCheck
+    make -C tiledb tests -j$NIX_BUILD_CORES
+    make -C tiledb test ARGS="-R '^unit_'" -R "test_assert"
+    make -C tiledb test ARGS="-R 'test_ci_asserts'"
+    runHook postCheck
   '';
 
   doCheck = true;
@@ -85,5 +116,4 @@ stdenv.mkDerivation rec {
     platforms = platforms.unix;
     maintainers = with maintainers; [ rakesh4g ];
   };
-
 }

